infra/lib/fhir-server-stack.ts

```diff
@@ -0,0 +1,273 @@
+    new CfnOutput(this, "FargateServiceARN", {
+      description: "Fargate Service ARN",
+      value: fargateService.service.serviceArn,
+    });
+    new CfnOutput(this, "DBClusterID", {
+      description: "DB Cluster ID",
+      value: dbCluster.clusterIdentifier,
+    });
+    new CfnOutput(this, "FHIRServerDBCluster", {
+      description: "FHIR server DB Cluster",
+      value: `${dbCluster.clusterEndpoint.hostname} ${dbCluster.clusterEndpoint.port} ${dbCluster.clusterEndpoint.socketAddress}`,
+    });
+  }
+
+  private setupDB(props: FHIRServerProps): {
+    dbCluster: rds.IDatabaseCluster;
+    dbCreds: { username: string; password: secret.Secret };
+  } {
+    // create database credentials
+    const dbClusterName = "fhir-server";
+    const dbName = props.config.dbName;
+    const dbUsername = props.config.dbUsername;
+    const dbPasswordSecret = new secret.Secret(this, "FHIRServerDBPassword", {
+      secretName: "FHIRServerDBPassword",
+      generateSecretString: {
+        excludePunctuation: true,
+        includeSpace: false,
+      },
+    });
+    const dbCreds = Credentials.fromPassword(
+      dbUsername,
+      dbPasswordSecret.secretValue
+    );
+    // aurora serverlessv2 db
+    const dbCluster = new rds.DatabaseCluster(this, "FHIR_DB", {
+      engine: rds.DatabaseClusterEngine.auroraPostgres({
+        version: rds.AuroraPostgresEngineVersion.VER_14_4,
+      }),
+      instanceProps: {
+        vpc: this.vpc,
+        instanceType: new InstanceType("serverless"),
+      },
+      credentials: dbCreds,
+      defaultDatabaseName: dbName,
+      clusterIdentifier: dbClusterName,
+      storageEncrypted: true,
+    });
+
+    const minDBCap = this.isProd(props) ? 2 : 1;
+    const maxDBCap = this.isProd(props) ? 8 : 2;
+    Aspects.of(dbCluster).add({
+      visit(node) {
+        if (node instanceof rds.CfnDBCluster) {
+          node.serverlessV2ScalingConfiguration = {
+            minCapacity: minDBCap,
+            maxCapacity: maxDBCap,
+          };
+        }
+      },
+    });
+
+    // add performance alarms for monitoring prod environment
+    if (this.isProd(props)) {
+      this.addDBClusterPerformanceAlarms(dbCluster, dbClusterName);
+    }
+    return {
+      dbCluster,
+      dbCreds: { username: dbUsername, password: dbPasswordSecret },
+    };
+  }
+
+  private setupFargateService(
+    props: FHIRServerProps,
+    dbCluster: rds.IDatabaseCluster,
+    dbCreds: { username: string; password: secret.Secret }
+  ): ecs_patterns.NetworkLoadBalancedFargateService {
+    // Create a new Amazon Elastic Container Service (ECS) cluster
+    const cluster = new ecs.Cluster(this, "FHIRServerCluster", {
+      vpc: this.vpc,
+    });
+
+    // Retrieve a Docker image from Amazon Elastic Container Registry (ECR)
+    const dockerImage = ecs.ContainerImage.fromEcrRepository(
+      ecr.Repository.fromRepositoryName(
+        this,
+        "FHIRServerECR",
+        props.config.fhirServerECRName
+      )
+    );
```

Comment:
why don't we just do what we do in the api stack?

https://github.com/metriport/metriport/blob/66a93687f221abe66ed89e9d018125f904fcfd42/infra/lib/api-stack.ts#L165-L167

this will build & upload to ECR for us

Reply:
Yeah, I started this on OSS then moved the infra here.

The main reason, IIRC, where:
1. CI/CD to fire the deployment comes from updates on this repo
1. Contents to be deployed come from this repo

We ended up pushing a docker image to ECR, so (2) doesn't play a role on this any more.

I'll think about it, maybe we can force a new deploy from this repo's CI/CD using AWS cli (below). But it would be odd to have a new PR on a diff repo just to make a release of this service, while we could have it all in one (following the idea of grouped repos, like we're doing with monorepos).

Also, I'm not sure just having the Fargate/ECS pointing to the `latest` ECR image will work. I just tried to make a new deployment here, and CDK didn't see any diff in task def, so didn't make a new deployment (nor restart of tasks) - I had to manually restart the service (below). Could need to make the task def to point to the current image's tag (`git rev-parse --short HEAD`), then this would go away (and only on edge cases where we need to make a deployment without a new commit we would need to manually restart it).

Thoughts?

For reference, since I just did this - AWS cli to force a new deployment/reload of existing ECS task def:

```shell
$ aws ecs update-service --force-new-deployment --service <SERVICE> --cluster <CLUSTER> --region <REGION>
```

Reply:
I missunderstood your message. Looking into it.

Reply:
Yeah, I started off with the idea of having infra on OSS and the code on FHIR server... then I though, "Ill build the image and push to ECR from the FHIR server's repo, and have the actual infra on the OSS".

When I moved the infra from OSS to FHIR server I forgot to question this and reuse the Docker Image construct straight on the CDK code... 🤦 

Adjusting...